samtranslator/utils/actions.py

```diff
@@ -0,0 +1,60 @@
+from abc import ABC, abstractmethod
+from typing import Any, Dict
+
+
+class Action(ABC):
+    """
+    Base class for Resolver function actions. Each Resolver function must subclass this,
+    override the , and provide a execute() method
+    """
+
+    @abstractmethod
+    def execute(self, template: Dict[str, Any]) -> Dict[str, Any]:
+        pass
+
+
+class ResolveDependsOn(Action):
+    DependsOn = "DependsOn"
+
+    def __init__(self, resolution_data: Dict[str, str]):
+        """
+        Initializes ResolveDependsOn. Where data necessary to resolve execute can be provided.
+
+        :param resolution_data: Extra data necessary to resolve execute properly.
+        """
+        self.resolution_data = resolution_data
+
+    def execute(self, template: Dict[str, Any]) -> Dict[str, Any]:
+        """
+        Resolve DependsOn when logical ids get changed when transforming (ex: AWS::Serverless::LayerVersion)
+
+        :param input_dict: Chunk of the template that is attempting to be resolved
+        :param resolution_data: Dictionary of the original and changed logical ids
+        :return: Modified dictionary with values resolved
+        """
+        # Checks if input dict is resolvable
+        if template is None or not self._can_handle_depends_on(input_dict=template):
+            return template
+        # Checks if DependsOn is valid
+        if not (isinstance(template[self.DependsOn], (list, str))):
+            return template
+        # Check if DependsOn matches the original value of a changed_logical_id key
+        for old_logical_id, changed_logical_id in self.resolution_data.items():
```

Comment:
Instead of looping through the dict, we can fetch the value directly from the dict.